buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy

```diff
@@ -162,71 +162,107 @@ class DistributedPerformanceTest extends ReportGenerationPerformanceTest {
+            "number": "921",
+            "status": "FAILURE",
+            "state": "finished",
+            "branchName": "master",
+            "href": "/app/rest/builds/id:14585813",
+            "webUrl": "https://builds.gradle.org/viewLog.html?buildId=14585813&buildTypeId=Gradle_Check_NoDaemon_Java8_Oracle_Windows_workers",
+            "statusText": "Gradle exception (new); exit code 1 (new)",
+            "buildType": {
+                "id": "Gradle_Check_NoDaemon_Java8_Oracle_Windows_workers",
+                "name": "Test Coverage - NoDaemon Java8 Oracle Windows (workers)",
+                "projectName": "Gradle / Check / Release Accept / Test Coverage - NoDaemon Java8 Oracle Windows",
+                "projectId": "Gradle_Check_NoDaemon_Java8_Oracle_Windows",
+                "href": "/app/rest/buildTypes/id:Gradle_Check_NoDaemon_Java8_Oracle_Windows_workers",
+                "webUrl": "https://builds.gradle.org/viewType.html?buildTypeId=Gradle_Check_NoDaemon_Java8_Oracle_Windows_workers"
+            },
+            "lastChanges": {
+                "change": [{
+                    "id": 476592,
+                    "version": "46ea4a59b549acea726dde8caa87307237a9679e",
+                    "username": "gary",
+                    "date": "20180730T194944+0000",
+                    "href": "/app/rest/changes/id:476592",
+                    "webUrl": "https://builds.gradle.org/viewModification.html?modId=476592&personal=false"
+                }],
+                "count": 1
+            },
+            ...
+        }
+        */
+
+        String workerBuildId = response.id
         cancellationToken.addCallback {
             cancel(workerBuildId)
         }
-        def scheduledChangeId = findLastChangeIdInXml(response.data)
+        def scheduledChangeId = findLastChangeIdInJson(response)
         if (lastChangeId && scheduledChangeId != lastChangeId) {
-            throw new RuntimeException("The requested change id is different than the actual one. requested change id: $lastChangeId in coordinatorBuildId: $buildId, actual change id: $scheduledChangeId in workerBuildId: $workerBuildId\nresponse: ${xmlToString(response.data)}")
+            throw new RuntimeException("The requested change id is different than the actual one. requested change id: $lastChangeId in coordinatorBuildId: $buildId, actual change id: $scheduledChangeId in workerBuildId: $workerBuildId\nresponse: $response")
         }
         scheduledBuilds.put(workerBuildId, scenario)
     }
 
     @TypeChecked(TypeCheckingMode.SKIP)
-    private static String xmlToString(xmlObject) {
-        if (xmlObject != null) {
-            try {
-                return XmlUtil.serialize(xmlObject)
-            } catch (e) {
-                // ignore errors
-            }
+    private CoordinatorBuild resolveCoordinatorBuild() {
+        if (buildId) {
+            Map response = httpGet(path: "builds/id:$buildId")
```

Comment:
Don't you need to add request content type `JSON` to this request?

Reply:
Then how do you make sure the server sends you json? Doing the same request in the old code did the same http request and that returned XML.

Reply:
I set `Accept` globally when creating `RESTClient`.